opentelekomcloud/services/ddm/resource_opentelekomcloud_ddm_instance_v1.go

```diff
@@ -0,0 +1,495 @@
+		nodesList = append(nodesList, node)
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("nodes", nodesList),
+	)
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceDdmInstanceV1Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	clientV1, err := common.ClientFromCtx(ctx, keyClientV1, func() (*golangsdk.ServiceClient, error) {
+		return config.DdmV1Client(config.GetRegion(d))
+	})
+	if err != nil {
+		return fmterr.Errorf(errCreationV1Client, err)
+	}
+
+	clientV2, err := common.ClientFromCtx(ctx, keyClientV2, func() (*golangsdk.ServiceClient, error) {
+		return config.DdmV2Client(config.GetRegion(d))
+	})
+	if err != nil {
+		return fmterr.Errorf(errCreationV2Client, err)
+	}
+
+	if d.HasChange("node_num") {
+		err = resourceDDMScaling(clientV1, clientV2, d, ctx)
+		if err != nil {
+			return fmterr.Errorf("error in DDM instance scaling: %w", err)
+		}
+	}
+
+	if d.HasChange("name") {
+		_, newNameRaw := d.GetChange("name")
+		newName := newNameRaw.(string)
+		log.Printf("[DEBUG] Renaming instance %s: %s", d.Id(), newName)
+		_, err = ddmv1instances.Rename(clientV1, d.Id(), newName)
+		if err != nil {
+			return fmterr.Errorf("error renaming DDM instance: %w", err)
+		}
+	}
+
+	if d.HasChange("security_group_id") {
+		_, newSecGroupRaw := d.GetChange("security_group_id")
+		newSecGroup := newSecGroupRaw.(string)
+		modifySecurityGroupOpts := ddmv1instances.ModifySecurityGroupOpts{
+			SecurityGroupId: newSecGroup,
+		}
+		_, err := ddmv1instances.ModifySecurityGroup(clientV1, d.Id(), modifySecurityGroupOpts)
+		if err != nil {
+			return fmterr.Errorf("error modifying DDM instance security group: %w", err)
+		}
+	}
+
+	if d.HasChange("password") {
+		clientV3, err := common.ClientFromCtx(ctx, keyClientV3, func() (*golangsdk.ServiceClient, error) {
+			return config.DdmV3Client(config.GetRegion(d))
+		})
+		if err != nil {
+			return fmterr.Errorf(errCreationV3Client, err)
+		}
+
+		_, err = accounts.ManageAdminPass(clientV3, d.Id(), accounts.ManageAdminPassOpts{
+			Name:     d.Get("username").(string),
+			Password: d.Get("password").(string),
+		})
+		if err != nil {
+			return fmterr.Errorf("error updating instance password: %w", err)
+		}
+	}
+
+	clientCtx := common.CtxWithClient(ctx, clientV1, keyClientV1)
+	return resourceDdmInstanceV1Read(clientCtx, d, meta)
+}
+
+func resourceDdmInstanceV1Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := common.ClientFromCtx(ctx, keyClientV1, func() (*golangsdk.ServiceClient, error) {
+		return config.DdmV1Client(config.GetRegion(d))
+	})
+	if err != nil {
+		return fmterr.Errorf(errCreationV1Client, err)
+	}
+
+	log.Printf("[DEBUG] Deleting Instance %s", d.Id())
+
+	deleteRdsData := d.Get("purge_rds_on_delete").(bool)
+	_, err = ddmv1instances.Delete(client, d.Id(), deleteRdsData)
```

Comment:
without default change to:
```
	purgeRds := false
	if v, ok := d.GetOk("purge_rds_on_delete"); ok {
		purgeRds = v.(bool)
	}
	_, err = ddmv1instances.Delete(client, d.Id(), purgeRds)
```

Reply:
Importer still throws error. It is set to false on default when state is created but in case this is added to true in test, the importer throws error. The only way i found that works is keeping the default but ignoring the attribute.

Reply:
Ah, yes i was wrong